soql-standalone-parser/src/main/scala/com/socrata/soql/ast/Select.scala

```diff
@@ -74,20 +101,21 @@ object Select {
 case class Select(
   distinct: Boolean,
   selection: Selection,
+  from: Option[TableName],
   joins: Seq[Join],
   where: Option[Expression],
   groupBys: Seq[Expression],
   having: Option[Expression],
   orderBys: Seq[OrderBy],
   limit: Option[BigInt],
   offset: Option[BigInt],
-  search: Option[String]) {
+  search: Option[String]) extends BinaryTree[Select] {
 
   private def toString(from: Option[TableName]): String = {
     if(AST.pretty) {
       val distinctStr = if (distinct) "DISTINCT " else ""
       val selectStr = Some(s"SELECT $distinctStr$selection")
-      val fromStr = from.map(t => s"FROM $t")
+      val fromStr = this.from.orElse(from).map(t => s"FROM $t")
```

Comment:
This `this` seems kinda unnecessary?

Reply:
this is this class from.  It is different than the parameter from.  This class from take precedence.  We actually want to remove the parameter from eventually but not yet.

Reply:
Ah. In that case, carry on. Also it's horrifying.